R/utils_plotting.R

```diff
@@ -398,23 +397,23 @@ NULL
 }
 
 .get_rect_args <- function(colour_by, alpha = 1, colour = "black"){
-    aes_args <- list()
+    aes_args <- aes()
     if (!is.null(colour_by)) {
-        aes_args$fill <- "colour_by"
+        aes_args$fill <- substitute(.data[["colour_by"]])
     }
-    new_aes <- do.call(aes_string, aes_args)
+    new_aes <- do.call(aes, aes_args)
     geom_args <- list(mapping = new_aes, alpha = alpha, colour = colour)
     return(list(args = geom_args))
 }
 
 .get_density_args <- function(colour_by, alpha = 0.65, colour = "black") {
     fill_colour <- TRUE
-    aes_args <- list()
+    aes_args <- aes()
     if (!is.null(colour_by)) {
-        aes_args$colour <- "colour_by"
-        aes_args$fill <- "colour_by"
+        aes_args$colour <- substitute(`colour_by`)
+        aes_args$fill <- substitute(`colour_by`)
     }
-    new_aes <- do.call(aes_string, aes_args)
+    new_aes <- do.call(aes, aes_args)
     geom_args <- list(mapping = new_aes,
```

Comment:
To get this working with as little modifications as possible (which is desirable), you can just add there

args <- lapply(args, function(x) if (!is.null(x)) sym(x))

Before do.call